File: problems/32/solution.java
// TreeNode type
// class TreeNode {
//     int val;
//     TreeNode left;
//     TreeNode right;
//     TreeNode(int x) { val = x; left = null; right = null; }
// }
class Solution {
    ArrayList<Integer> result = new ArrayList<Integer>();

    public ArrayList<Integer> inorderTraversal(TreeNode root) {
        // Implement the solution
        _inorder(root);
        return result;
    }


    public void _inorder(TreeNode root) {
        if (root == null)
            return;

        _inorder(root.left);
        result.add(root.val);
        _inorder(root.right);
    }

}

File: problems/32/solution.c
// struct node {
// 	  int val;
// 	  struct node* left;
//    struct node* right;
// };
// typedef struct node TreeNode;

// Please refrain from using print statements

const int MAXN = 100000;

void inorder(TreeNode* root, int* result, int* length) {
    if (!root)
        return;
    inorder(root->left, result, length);
    result[*length] = root->val;
    *length = *length + 1;
    inorder(root->right, result, length);
}

int* inorderTraversal(TreeNode* root, int* length) {
    int* result = (int*) malloc((sizeof (int)) * MAXN);
    *length = 0;
    inorder(root, result, length);
    return result;
}
